### DISCUSSION TITLE
[Question]'Articulation' object has no attribute '_data'.

### DISCUSSION THREAD

**Discussion by tesollo-admin** [isaac-sim/IsaacLab#623] [Q&A] [answered]
Hi, everyone.

I configured my own robot using `ArticulationCfg` and tried to get the `default_joint_pos` or `joint_vel` from `robot.data.default_joint_pos.clone()`. However, I encountered the following error:
```javascript
joint_pos = robot.data.default_joint_pos.clone()
  File "./IsaacLab/source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py", line 107, in data
    return self._data
AttributeError: 'Articulation' object has no attribute '_data'. Did you mean: 'data'?
```
But, I haven't seen the issue before.

Does anyone know the solutions about this?

Thank you in advance.

**Comment by tesollo-admin** [comment]
Solved with setting the default prim to the robot as commented by @Overdamp .
Thanks!

**Reply by Mayankm96** [reply]
Probably before this bug another bug happened where initialization callback failed. Can you share the log file?

**Comment by Overdamp** [comment, marked as answer]
![Screenshot from 2024-07-19 15-40-13](https://github.com/user-attachments/assets/adfb0f60-279a-4b0f-8e57-9ab82d58a130)
Same problem.I have already fix the problem by myself by setting the default prim to the robot.

**Reply by EricPedley** [reply]
~~How would you do this when creating the scene programatically through a `InteractiveSceneCfg` object like in this [tutorial](https://isaac-sim.github.io/IsaacLab/main/source/tutorials/03_envs/create_manager_base_env.html)?~~
Nevermind, it seems like for the example I'm trying to run, the robot is already the default prim when I just open the usd file on its own. However, I don't get this error when I run my custom script based on the tutorial with a different robot, so I'm giving up investigating further. (https://github.com/EricPedley/so101/blob/eb0b7d4d134682a6f12c0bdc0194fbc6096165cd/isaac_lab_main.py)
